### PR TITLE
Splash screen and navigation

### PR DESCRIPTION
**Pull request by ravishrajput**

### CODE REVIEW

File: app/src/main/java/com/remindme/app/screens/TimelineScreen.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.sp
+import androidx.navigation.NavController
+import androidx.navigation.compose.rememberNavController
+import com.remindme.app.R
+import com.remindme.app.ui.theme.RemindMeTheme
+import com.remindme.app.ui.theme.Teal700
+
+@Composable
+fun ShowTimelineScreen(navController: NavController) {
```
Comment (parmeshtoyou) [inline app/src/main/java/com/remindme/app/screens/TimelineScreen.kt:37]: is this the screen which is going to display items or cards which are going to expire soon?

File: app/src/main/java/com/remindme/app/screens/splash/SplashViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  override fun processAction(action: SplashViewAction) {
+    when (action) {
+      is SplashViewAction.Launch -> fetchRequiredData()
+    }
+  }
+
+  private fun fetchRequiredData() {
+    viewModelScope.launch {
+      // execute API call or process requirements for loading next screen
```
Comment (parmeshtoyou) [inline app/src/main/java/com/remindme/app/screens/splash/SplashViewModel.kt:36]: which API we are planning to call here? Asking just for curiosity.

File: app/src/main/java/com/remindme/app/viewmodel/BaseViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+  protected abstract val initialViewState: ViewState
+  protected abstract fun processAction(action: ViewAction)
+
+  protected var lastViewState: ViewState = initialViewState
+
+  private val _viewState = MutableStateFlow(initialViewState)
+  val viewState = _viewState.asStateFlow()
+
+  private val _viewEvents = Channel<ViewEvent?>(Channel.BUFFERED)
```
Comment (parmeshtoyou) [inline app/src/main/java/com/remindme/app/viewmodel/BaseViewModel.kt]: in which state ViewEvent can be null?

**Review by parmeshtoyou — APPROVED** [review]

File: app/src/main/java/com/remindme/app/screens/TimelineScreen.kt (hunk 1 above)
Comment (ravishrajput) [reply app/src/main/java/com/remindme/app/screens/TimelineScreen.kt:37]: yes

File: app/src/main/java/com/remindme/app/screens/splash/SplashViewModel.kt (hunk 2 above)
Comment (ravishrajput) [reply app/src/main/java/com/remindme/app/screens/splash/SplashViewModel.kt:36]: not sure for now, but may be firebase sync once we decide app architecture.

File: app/src/main/java/com/remindme/app/viewmodel/BaseViewModel.kt (hunk 3 above)
Comment (ravishrajput) [reply app/src/main/java/com/remindme/app/viewmodel/BaseViewModel.kt]: made non nullable.

**Pull request merged**